lms/static/js/ccx/collection/schedule_collection.js

```diff
@@ -0,0 +1,275 @@
+
+                                    _.each(subSection.children, function (__, verticalIndex) {
+                                        // update start and due dates of verticals
+                                        tree[chapterIndex].children[subSectionIndex].
+                                            children[verticalIndex].start = unit.start;
+
+                                        tree[chapterIndex].children[subSectionIndex].
+                                            children[verticalIndex].due = unit.due;
+                                    });
+                                    return true;
+                                }
+                            });
+                            return true;
+                        }
+                    });
+                },
+
+                pruned: function (tree, filter) {
+                    var self = this;
+                    return tree.filter(filter)
+                        .map(function (node) {
+                            var copy = {};
+                            $.extend(copy, node);
+                            if (node.children) {
+                                copy.children = self.pruned(node.children, filter);
+                            }
+                            return copy;
+                        }).filter(function (node) {
+                            return node.children === undefined || node.children.length;
+                        });
+                },
+
+                treeFilter: function (unit) {
+                    return !unit.hidden;
+                },
+
+                formFilter: function (unit) {
+                    return unit.hidden || unit.category !== 'vertical';
+                },
+
+                changeVisibilityOfUnitInSchedule: function (units, applyVisibility) {
+                    var self = this;
+                    units.map(function (unit) {
+                        applyVisibility(unit);
+                        if (unit !== undefined && unit.children !== undefined) {
+                            self.changeVisibilityOfUnitInSchedule(unit.children, applyVisibility);
+                        }
+                    });
+                },
+
+                hide: function (unit) {
+                    if (unit !== undefined) {
+                        unit.hidden = true;
+                    }
+                },
+
+                show: function (unit) {
+                    if (unit !== undefined) {
+                        unit.hidden = false;
+                    }
+                },
+
+                validUnitDates: function (start, due, unit) {
+                    var errorMessage;
+                    // Start date is compulsory and due date is optional.
+                    if (_.isEmpty(start) && !_.isEmpty(due)) {
+                        errorMessage = StringUtils.interpolate(
+                            gettext(
```

Comment:
Do not break JavaScript strings into multiples lines when passing to gettext-- it breaks the scraping. If you are getting a JSHint violation on the length of the line, suppress the warning for this line only.